index.d.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * All exports from TS source files (where the implementation is actualy done in TS).
+ */
+export * from './target/types/type_exports';
```

Comment:
Ok, I've updated the build to build this project, but in the repo root rather than the build root. The reason for this is we build the typescript files from the repo root in x-pack so the types will need to be present there. I've also added an alias so you can now do this in any x-pack code:

```ts
import { Legacy } from 'kibana'
```